packages/sanity/src/_internal/cli/actions/graphql/extractFromSanitySchema.ts

```diff
@@ -393,8 +393,24 @@ export function extractFromSanitySchema(
       return {type: getTypeName(candidates[0].type.name), ...base}
     }
 
-    const unionDefinition = getUnionDefinition(candidates, def, {grandParent: parent})
-    return {...unionDefinition, ...base}
+    const allTypeNames = candidates.map((c) => getTypeName(c.type.name))
+    const targetTypes = [...new Set(allTypeNames)].sort()
+    const name = targetTypes.join('Or')
+
+    // Register the union type if we haven't seen it before
+    if (!unionTypes.some((item) => item.name === name)) {
+      unionTypes.push({
+        kind: 'Union',
+        name,
+        types: targetTypes,
+      })
+    }
+
+    return {
+      type: name,
+      // references: targetTypes,
```

Comment:
Would we like to have this? It seems like what we would like to do, based on the usage in getUnionDefinition, but this implementation adds quite a bit of content to our testing snapshots.

Reply:
Can you make it "optional"?
```
const def = {type: name, ...base}
if (targetTypes !== undefined) {
  def.references = targetTypes
}
```